apollo-router/src/query_planner/bridge_query_planner.rs

```diff
@@ -353,11 +389,18 @@ mod tests {
             .unwrap_err();
 
         match err {
-            QueryPlannerError::PlanningErrors(plan_errors) => {
+            QueryPlannerError::SpecError(spec_error) => {
+                // TODO(@goto-bus-stop): it's not useful to do this here right? should be part of
+                // SpecError maybe?
```

Comment:
why do we need to explicitely create the usage reporting structure here? It's supposed come from the error returned by the planner